hazelcast-jet-core/src/main/java/com/hazelcast/jet/function/DistributedFunctions.java

```diff
@@ -63,12 +60,14 @@ private DistributedFunctions() {
     }
 
     /**
-     * Returns a function that always evaluates to the {@link #CONSTANT_KEY}.
-     * This is useful as a key extractor in group-by operations where no
-     * classification by key is desired.
+     * Returns a function that always evaluates to the given {@code key}. This
+     * is useful as a key extractor in group-by operations where no
+     * classification by key is desired; in other words, a global group is
+     * used. Choose a random {@code key} so that each global group is handled
+     * by a random member.
      */
     @Nonnull
-    public static <T> DistributedFunction<T, String> constantKey() {
-        return t -> CONSTANT_KEY;
+    public static <T, K> DistributedFunction<T, K> constantKey(K key) {
```

Comment:
does this need to be part of public API? I think it would be better to move it to `ConstantFunction`

Reply:
Yes, it doesn't. Pipeline API uses it internally and it's not needed for core api - the user can use `allToOne` and appropriate key extractor in the downstream vertex.